### PR TITLE
Query Title block: Rely on the editor store to apply the right archive title placeholder.

### PR DESCRIPTION
**Pull request by youknowriad**
This is the first step to solve this but #59082 

## What?

This PR refactors the way the Query Title block retrieves the current template in order to change the placeholder contextually. Instead of relying on a random block editor setting, it relies on the actual template used in the "core/editor" store directly.

The reason we can do this is because some blocks of block-library are WP specific and can actually make use of the editor package. It's the case for the Query Title block.

## Testing Instructions

1- Create tag archive block template
2- Make the sure the title shows the right placeholder: "Tag: name".

### CODE REVIEW

File: packages/block-library/src/query-title/use-archive-label.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+			( currentPostType === 'wp_template' ? getCurrentPostId() : null );
+
+		return templateId
+			? select( coreStore ).getEditedEntityRecord(
+					'postType',
+					'wp_template',
+					templateId
+			  )?.slug
+			: null;
+	}, [] );
```
Comment (youknowriad) [inline packages/block-library/src/query-title/use-archive-label.js:29]: Above this line is the new addition. The rest is the same as before.

**Review by jameskoster — APPROVED** [review]
> 1- Create tag archive block template
> 2- Make the sure the title shows the right placeholder: "Tag: name".

👍

**Review by ntsekouras — COMMENTED** [review]
Looks good. The only visible change though is that since the hook is inside the block's edit function, there is the quick flash updates of titles, when the selector has resolved. Should we wait for the resolution? 🤔

**Comment by youknowriad** [comment]
I don't really understand why it's different in terms of loading than trunk, we're calling the exact same selectors. 🤔

**Comment by youknowriad** [comment]
I see now, It's actually the same as before, only difference is the "global loading" was taking the resolver into consideration and in this PR it's not. I'm guessing because the block is rendered after some "ticks". 

Going to merge now I guess but we can look at improving that if needed.

**Pull request merged**